### PR TITLE
[#129] Implemented new method of retrieving data from database and added assertions for unit tests. 

### PR DESCRIPTION
**Pull request by Mstc08**
[#129]
-Data are now retrieved using IDs.
-Added assertions in unit test to ensure the proper behavior of the unit test.
-Added method to drop collections after the specs have finished executing.
-The following files have been modified:
companyAccountSpec, editUserPageSpec, indexSpec, jobPageSpec, jobPostingPageSpec, loginRegisterSpec, questionPromptSpec, questionSpec, userPageSpec

### CODE REVIEW

File: routes/question.js [hunk 1]
```diff
@@ -319,4 +319,6 @@ router.patch("/:replyId/updownedReply", function(req, res, next) {
         });
 });
 
-module.exports = router;
\ No newline at end of file
+module.exports = {router,getPropertyWithTag};
+// module.exports = router;
```
[the remark this answers is not in the document]
Comment (Mstc08) [reply routes/question.js]: Fixed

File: test/companyAccountSpec.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
 const mongoose = require('mongoose');
-const Schema = mongoose.Schema;
+const Schema = mongoose.Schema;		
+var expect = require('chai').expect;
 
 const testCompanySchema = mongoose.Schema({
+  _id: mongoose.Schema.Types.ObjectId,
```
[the remark this answers is not in the document]
Comment (Mstc08) [reply test/companyAccountSpec.js]: Fixed

File: test/companyAccountSpec.js [hunk 3]
```diff
@@ -35,26 +39,76 @@ describe('Connecting to database for company account', function() {
...
+
     it('Should retrieve a company from test database', function(done) {
-        testCompanyAccount.find({username: 'Company username'}, (err, name) => {
-          if(err) {throw err;}
-          if(name.length === 0) {throw new Error('No data!');}
-          done();
+      testCompanyAccount.findById(companyID)
+      .exec()
+      .then(function(doc){
+        //This is an example of asserting, here we are asserting that the company found is the one we indeed created at the beginning
```
[the remark this answers is not in the document]
Comment (Mstc08) [reply test/companyAccountSpec.js]: Fixed

File: test/editUserPageSpec.js [hunk 4]
```diff
@@ -1,7 +1,9 @@
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
+var expect = require('chai').expect;
 
 const testEditUserSchema = mongoose.Schema({
+  _id: mongoose.Schema.Types.ObjectId,
```
[the remark this answers is not in the document]
Comment (Mstc08) [reply test/editUserPageSpec.js]: Fixed

File: test/indexSpec.js [hunk 5]
```diff
@@ -1,10 +1,12 @@
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
+var expect = require('chai').expect;
 const Reply = require("../models/reply");
 
 const reply = mongoose.model('reply', Reply.schema);
 
 const testQuestionSchema = mongoose.Schema({
+    _id: mongoose.Schema.Types.ObjectId,
```
[the remark this answers is not in the document]
Comment (Mstc08) [reply test/indexSpec.js]: Fixed

File: test/jobPageSpec.js [hunk 6]
```diff
@@ -1,21 +1,22 @@
...
 
 const testJobPageSchema = mongoose.Schema({
-    _id: mongoose.Schema.Types.ObjectId,
-    title : {type: String, required: true},
-    company: {type: String, required: true},
-    date :  String,
-    link : {type: String, required: true},
-    author : {type: String, required: true},
-    location : {type: String, required: true}
+  _id: mongoose.Schema.Types.ObjectId,
```
[the remark this answers is not in the document]
Comment (Mstc08) [reply test/jobPageSpec.js]: Fixed

File: test/jobPostingPageSpec.js [hunk 7]
```diff
@@ -1,21 +1,22 @@
...
 
 const testJobPostingSchema = mongoose.Schema({
-    _id: mongoose.Schema.Types.ObjectId,
-    title : {type: String, required: true},
-    company: {type: String, required: true},
-    date :  String,
-    link : {type: String, required: true},
-    author : {type: String, required: true},
-    location : {type: String, required: true}
+  _id: mongoose.Schema.Types.ObjectId,
```
[the remark this answers is not in the document]
Comment (Mstc08) [reply test/jobPostingPageSpec.js]: Fixed

File: test/loginRegisterSpec.js [hunk 8]
```diff
@@ -1,7 +1,9 @@
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
+var expect = require('chai').expect;
 
 const testLoginRegisterSchema = mongoose.Schema({
+  _id: mongoose.Schema.Types.ObjectId,
```
[the remark this answers is not in the document]
Comment (Mstc08) [reply test/loginRegisterSpec.js]: Fixed

File: test/questionPromptSpec.js [hunk 9]
```diff
@@ -18,7 +19,7 @@ const testQuestions = mongoose.model('testQuestions1', testQuestionSchema);
 
 describe('Connecting to database for question prompt page', function() {
   before(function (done) {
-    mongoose.connect('mongodb://soen341:soen341@soen341-shard-00-00-ruxjj.mongodb.net:27017,soen341-shard-00-01-ruxjj.mongodb.net:27017,soen341-shard-00-02-ruxjj.mongodb.net:27017/test?ssl=true&replicaSet=SOEN341-shard-0&authSource=admin');
+    mongoose.connect("mongodb://soen341:soen341@soen341-shard-00-00-ruxjj.mongodb.net:27017,soen341-shard-00-01-ruxjj.mongodb.net:27017,soen341-shard-00-02-ruxjj.mongodb.net:27017/test?ssl=true&replicaSet=SOEN341-shard-0&authSource=admin");
```
[the remark this answers is not in the document]
Comment (Mstc08) [reply test/questionPromptSpec.js]: Fixed

File: test/questionSpec.js [hunk 10]
```diff
@@ -58,82 +97,224 @@ describe('Database Tests for question page', function() {
...
+            })
+            .catch(function(err){
+                console.log(err);
+                res.status(500).json({
+                    error:err
+                });
+            })
+        });
+
+        //IMPORTANT!! Here modify the tests below to conform to the new way
```
[the remark this answers is not in the document]
Comment (Mstc08) [reply test/questionSpec.js]: These comments are left there to allow Panorea to know how to modify her tests to conform to the new way

[the remark this answers is not in the document]
Comment (kjanei) [reply test/questionSpec.js]: Okay, but they should be removed once @panoreak makes her changes.

**Pull request merged**